cmd/globalAlignment/globalAlignment.go

```diff
@@ -8,17 +8,40 @@ import (
 	"github.com/vertgenlab/gonomics/fileio"
 	"github.com/vertgenlab/gonomics/simpleGraph"
 	"log"
+	"strings" //raven added this line for CountSeqIdx
+	"os" //raven added this line for MSA Fasta output
 )
 
+//raven edited this block to specify only 1 sequnce is expected in each fasta file
 func usage() {
-	fmt.Print("./globalAlignment - chelsea's global alignment \n")
+	fmt.Print(
+	"./globalAlignment - chelsea's global alignment\n" +
+	"Usage:\n" +
+	" Align 2 .fasta files, each with only 1 sequence\n" +
+	"options:\n")
 	flag.PrintDefaults()
 }
 
+//raven wrote this block to count sequences based on the Read function in gonomics/fasta/fasta.go
+//maybe there are existing functions that do this already? I didn't find an existing function that returns seqIdx
+func CountSeqIdx(filename string) int {
+	var line string
+	var seqIdx int = 0 //I know in Read seqIdx is int64 and starts with -1, but I am using it differently here
+	var doneReading bool = false
+	file := fileio.EasyOpen(filename)
+	defer file.Close()
```

Comment:
Just curious, is there a reason other than stylistic to defer file.close() rather than just adding it to the end of this function?  Is it maybe a best practice thing? Or maybe there is some computational benefits too?

Reply:
Actually I think it's the reverse lol. I think a lot of people use `defer file.Close()` (myself included) out of habit and convenience, but in reality it is best to call `file.Close()` after you are done working with the file so that you can do error checking. The `Close()` function does have an error return, but by deferring the function you supress the error making it possible for some errors to go unnoticed.

Reply:
I called file.Close() to try it out. Thank you!